pkg/informer/informer.go

```diff
@@ -44,6 +101,154 @@ func InformerForResource(res schema.GroupVersionResource) (GenericInformer, erro
+						// fmt.Println("deny group, api, kind, resources", group, api, kind, deny.Resources[i])
+						boolVar = false
+					}
+
+				}
+			} else {
+				// if api != "*" && deny.Resources[i] != "*" {
+				if group == api && kind == deny.Resources[i] {
+					// fmt.Println("deny group, api, kind, resources", group, api, kind, deny.Resources[i])
+					boolVar = false
+				}
+				// }
+			}
+		}
+	}
+
+	// if allowedlist is empty allow all resources, otherwise if * allow all groups specific
+	// resources if not * allow specific resources to specific groups:
+	if len(allowedList) == 0 {
+		boolVar = true
+	} else {
+		// fmt.Println("In-coming ApiResource", kind)
+		for i, allow := range allowedList {
+			for _, api := range allow.ApiGroups {
+				if api == "*" && allow.Resources[i] != "*" { //all apigroups & specific resources
+					if allow.Resources[i] == kind {
+						// fmt.Println("allow group, api, kind, resources", group, api, kind, allow.Resources[i])
+						boolVar = true
+
+					} else {
+						if api != "*" && allow.Resources[i] == "*" { // specific apigroups & all resources
+							// fmt.Println("allow group, api, kind, resources", group, api, kind, allow.Resources[i])
+							boolVar = true
+						}
+					}
+
+				} else {
+					// if api != "*" && allow.Resources[i] != "*" { //specific apigroups and resources
+					if group == api && kind == allow.Resources[i] {
+						// fmt.Println("allow group, api, kind, resources", group, api, kind, allow.Resources[i])
+						boolVar = true
+					}
+					// }
+				}
+			}
+		}
+	}
+
+	return boolVar
+}
+
+// Returns a map containing all the GVRs on the cluster of resources that support WATCH (ignoring clusters and events).
+func SupportedResources(discoveryClient *discovery.DiscoveryClient) (map[schema.GroupVersionResource]struct{}, error) {
+	// Next step is to discover all the gettable resource types that the kuberenetes api server knows about.
+	supportedResources := []*machineryV1.APIResourceList{}
+
+	// List out all the preferred api-resources of this server.
+	apiResources, err := discoveryClient.ServerPreferredResources() //<--here we can look into this list and have preferred versions
+	if err != nil && apiResources == nil {                          // only return if the list is empty
+		return nil, err
+	} else if err != nil {
+		glog.Warning("ServerPreferredResources could not list all available resources: ", err)
+	}
+
+	// create client to get configmap
+	config := config.GetKubeConfig() //can't err here?
+	clientset, err := kubernetes.NewForConfig(config)
+
+	//locate the allow-deny ConfigMap:
+	var cm *v1.ConfigMap
+	if cm, err = clientset.CoreV1().ConfigMaps("open-cluster-management").Get(contextVar, "allowdeny-config", metav1.GetOptions{}); err != nil {
+		glog.Warning("Can't find allow/deny ConfigMap", err)
+	}
+
+	var deniedList []DeniedResources
+	var allowedList []AllowedResources
+
+	//parse config:
+	allowedList, deniedList = GetAllowDenyData(cm)
```

Comment:
You can use `:=` instead of declaring the var.
```suggestion
	//parse config:
	allowedList, deniedList := GetAllowDenyData(cm)
```